src/ashley/models.py

```diff
@@ -151,6 +152,15 @@ def sync_user_groups(self, user, roles: List[str]) -> None:
                 logger.debug("Add user %s to group %s", user, group)
                 user.groups.add(group)
 
+    def is_user_instructor(self, user) -> bool:
```

Comment:
I'm not confortable with such a specific method in the `LTIContext` model, as it adds the interpretation of the LTI roles to this model, in addition to reflecting their composition.

What do you think about adding a method `get_user_roles(self, user) -> List[str]` here, and handle the specific detection of the "instructor" role outside of this model ?